### PR TITLE
PPTP-1659: Do you want to start your return?

### PR DESCRIPTION
**Pull request by C-Thomp**
- Start date returns page /start-date created with tests for controller, form and the view.
- Obligations api connected up.

**NOTE:** ticket was to create two pages, /start-date-not-recognised. This page hasn't been completed yet due to shift in print priorities.

#### Check list 
 - [x] `./precheck` was executed (Integration/Component/Unit tests)
 - [x] `sbt scalafmt test:scalafmt` was executed 
 - [ ] Required Environment Config has been amended/added
 - [ ] Links to dependencies have been included (BE/FE work)
 - [ ] User Acceptance Tests (UAT) were run locally and have passed.
 - [ ] No Accessibility errors/warnings reported by Wave

### CODE REVIEW

File: test/uk/gov/hmrc/plasticpackagingtax/returns/controllers/returns/StartDateReturnControllerTest.scala [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        authorizedUser()
+        when(page.apply(any(), refEq(defaultObligation))(any(), any())).thenReturn(HtmlFormat.empty)
+        val req: Request[AnyContentAsFormUrlEncoded] = postRequest
+          .withFormUrlEncodedBody(FieldKey -> Yes)
+          .withCSRFToken
+
+        val result: Future[Result] = controller.submit()(req)
+
+        //todo: update when redirect pages are created
+        redirectLocation(result) mustBe Some(homeRoutes.HomeController.displayPage().url)
```
Comment (RyanKellyHMRC) [inline test/uk/gov/hmrc/plasticpackagingtax/returns/controllers/returns/StartDateReturnControllerTest.scala:79]: These pages should be available to be wired in now.

**Review by RyanKellyHMRC — CHANGES_REQUESTED** [review]
Looking good. There are few smaller changes worth making in my opinion and some bits which can be wired in now.

File: app/uk/gov/hmrc/plasticpackagingtax/returns/controllers/returns/StartDateReturnController.scala [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+@Singleton
+class StartDateReturnController @Inject() (
+  authenticate: AuthAction,
+  journeyAction: JourneyAction,
+  mcc: MessagesControllerComponents,
+  view: start_date_returns_page
+)(implicit ec: ExecutionContext)
+    extends FrontendController(mcc) with I18nSupport {
+
+  def displayPage(): Action[AnyContent] = //TODO: which dates to populate title message?
```
[the remark this answers is not in the document]
Comment (liampan) [reply app/uk/gov/hmrc/plasticpackagingtax/returns/controllers/returns/StartDateReturnController.scala:41]: Yes, its something that probably needs checked for the header section component thats already in main now. Unless you know thats right?

File: app/uk/gov/hmrc/plasticpackagingtax/returns/models/request/JourneyAction.scala [hunk 3]
```diff
@@ -63,13 +72,14 @@ class JourneyAction @Inject() (returnsConnector: TaxReturnsConnector, auditor: A
...
-                                              toDate = LocalDate.parse("2022-06-30"),
-                                              dueDate = LocalDate.parse("2022-06-30"),
-                                              periodKey = "22AP"
-            )
-            returnsConnector.create(TaxReturn(id = id, obligation = oldestObligation))
+
+            obligationsConnector.get(id).flatMap { openObligations =>
+              val oldestObligation = openObligations
+                .nextObligationToReturn
+                .getOrElse(throw new IllegalStateException(s"No Obligation for return id:$id"))
```
[the remark this answers is not in the document]
Comment (RyanKellyHMRC) [reply app/uk/gov/hmrc/plasticpackagingtax/returns/models/request/JourneyAction.scala:79]: Actually, we had to make `obligation` optional in order to cater for existing in-flight returns which won't have one (these are created when the dashboard is accessed which happens after a successful registration). I don't think all users will have an outstanding obligation when they hit the dashboard. Can we remove the throwing of `IllegalStateException` and just set as `None` if there isn't one? We could then (probably a future ticket if we decide to bother) just tell the user that they do not have any open obligations if they try to access this flow - I imagine we would not present a link on the dashboard in this context.

File: conf/messages.en [hunk 4]
```diff
@@ -26,9 +26,6 @@ month.10 = October
 month.11 = November
 month.12 = December
 
-general.true = Yes
```
Comment (liampan) [inline conf/messages.en:29]: removing duplicates. lines 13, 14

File: test/uk/gov/hmrc/plasticpackagingtax/returns/models/request/JourneyActionSpec.scala [hunk 5]
```diff
@@ -58,6 +58,7 @@ class JourneyActionSpec extends ControllerSpec {
         given(mockTaxReturnsConnector.find(refEq(taxReturnId))(any[HeaderCarrier])).willReturn(
           Future.successful(Right(Option(aTaxReturn(withId(taxReturnId)))))
         )
+        mockGetObligation()
```
Comment (RyanKellyHMRC) [inline test/uk/gov/hmrc/plasticpackagingtax/returns/models/request/JourneyActionSpec.scala:61]: If you used an `afterEach()` hook and reset mocks there then you could do "common" mock setup in `beforeEach()`. No need for a change - just a suggestion for the future.

File: app/uk/gov/hmrc/plasticpackagingtax/returns/models/request/JourneyAction.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (RyanKellyHMRC) [reply app/uk/gov/hmrc/plasticpackagingtax/returns/models/request/JourneyAction.scala]: I think this is going to be a problem but we can address on a future ticket.

File: app/uk/gov/hmrc/plasticpackagingtax/returns/views/returns/start_date_returns_page.scala.html [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+){
+    @errorSummary(form.errors)
+
+    @formHelper(action = controllers.returns.routes.StartDateReturnController.submit(), 'autoComplete -> "off") {
+
+        @govukRadios(Radios(
+            fieldset = Some(Fieldset(
+                legend = Some(Legend(
+                    content = Text(heading),
+                    classes = "govuk-fieldset__legend govuk-fieldset__legend govuk-fieldset__legend--l",
```
[the remark this answers is not in the document]
Comment (RyanKellyHMRC) [reply app/uk/gov/hmrc/plasticpackagingtax/returns/views/returns/start_date_returns_page.scala.html]: This has been done.

**Review by RyanKellyHMRC — APPROVED** [review]
Good work @C-Thomp. Thanks for making the suggested changes.

**Pull request merged**